File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    li = []
    for i in range(len(prices)):
        num = 0
        for j in range(i+1, len(prices)):
            num+=1
            if prices[i] > prices[j]:
                break
    
        li.append(num)
        
        
    return li

File: 프로그래머스/2/42584. 주식가격/주식가격.cpp
#include <string>
#include <vector>
#include <stack>

using namespace std;

vector<int> solution(vector<int> prices) {
    int n = prices.size();
    vector<int> answer(n, 0);
    stack<int> s;
    
     for (int i = 0; i < n; ++i) {
        while (!s.empty() && prices[i] < prices[s.top()]) {
            int topIndex = s.top();
            s.pop();
            answer[topIndex] = i - topIndex;
        }
        s.push(i); 
    }
    
    while (!s.empty()) {
        int topIndex = s.top();
        s.pop();
        answer[topIndex] = n - 1 - topIndex;
    }
    
    return answer;
}